Engine/Visa.cs

```diff
@@ -4,746 +4,1043 @@
+        /// <summary>Parse resource string to get extended interface information</summary>
+        int viParseRsrcEx(int sesn, string desc, ref short intfType, ref short intfNum, StringBuilder rsrcClass, StringBuilder expandedUnaliasedName, StringBuilder aliasIfExists);
+        /// <summary>Open session</summary>
+        int viOpen(int sesn, string viDesc, int mode, int timeout, out int vi);
+        /// <summary>Close session</summary>
+        int viClose(int vi);
+        /// <summary>Get attribute, returning a byte</summary>
+        int viGetAttribute1(int vi, int attrName, out byte attrValue);
+        /// <summary>Get attribute, filling a pre-allocated StringBuilder buffer</summary>
+        int viGetAttribute2(int vi, int attrName, StringBuilder attrValue);
+        /// <summary>Get attribute, returning an int</summary>
+        int viGetAttribute3(int vi, int attrName, out int attrValue);
+        /// <summary>Set attribute, providing a byte value</summary>
+        int viSetAttribute1(int vi, int attrName, byte attrValue);
+        /// <summary>Set attribute, providing an int value</summary>
+        int viSetAttribute2(int vi, int attrName, int attrValue);
+        /// <summary>Get status code description</summary>
+        int viStatusDesc(int vi, int status, StringBuilder desc);
+        /// <summary>Enable event</summary>
+        int viEnableEvent(int vi, int eventType, short mechanism, int context);
+        /// <summary>Disable event</summary>
+        int viDisableEvent(int vi, int eventType, short mechanism);
+        /// <summary>Install handler</summary>
+        int viInstallHandler(int vi, int eventType, viEventHandler handler, int UserHandle);
+        /// <summary>Uninstall handler</summary>
+        int viUninstallHandler(int vi, int eventType, viEventHandler handler, int userHandle);
+        /// <summary>Read data from device</summary>
+        unsafe int viRead(int vi, ArraySegment<Byte> buffer, int count, out int retCount);
+        /// <summary>Write data to device</summary>
+        unsafe int viWrite(int vi, ArraySegment<Byte> buffer, int count, out int retCount);
+        /// <summary>Read status byte</summary>
+        int viReadSTB(int vi, ref short status);
+        /// <summary>Clear a device</summary>
+        int viClear(int vi);
+        /// <summary>Lock resource</summary>
+        int viLock(int vi, int lockType, int timeout, string requestedKey, StringBuilder accessKey);
+        /// <summary>Unlock resource</summary>
+        int viUnlock(int vi);
+        /// <summary>Event handler prototype</summary>
+        delegate int viEventHandler(int vi, int eventType, int context, int userHandle);
+    }
 
-        Func<IntPtr> LoadLib;
-        Func<IntPtr, string, int, IntPtr> LoadSym;
+    /// <summary>
+    /// Attempt to load VISA libraries from the system
+    /// </summary>
+    public class VisaLibraryLoader : IVisa, IVisaProvider
```

Comment:
I think this can be internal. OpenTAP sets PluginAssembly attribute so internal types are also shown as plugins.